examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+    "* observe a traitlet and call python function\n",
+    "* observe a traitlet and call JavaScript function\n",
+    "* SOON: call python functions from JavaScript\n",
+    "* loading components from external files\n",
+    "* enable hot reloading  \n",
+    "* enable autocompletion in IDEs\n",
+    "* print a message at class initialization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "06cb067d-0aea-4851-827d-d07f22466693",
+   "metadata": {},
+   "source": [
+    "First, we will use the **`%react` magic**  from ipyreact.  \n",
+    "The following line registers the cellmagic:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de17b188",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%pip install ipyreact\n",
+    "# This line is for JupyterLite (in case that this takes long, restart the kernel and run this cell again)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d85f911",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext ipyreact"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c70f7253-98d5-4131-bf6b-1ac3433e2985",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%react\n",
+    "\n",
+    "import * as React from \"react\";\n",
+    "\n",
+    "export default function MyButton() {\n",
+    "    return ( < button > X < /button>);\n",
+    "    }"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4b51d34f-8abc-4b7b-936e-6d582a1b64fe",
+   "metadata": {},
+   "source": [
+    "This works like a charm!  \n",
+    "Next, we **convert this into a widget.**  \n",
+    "For that, we need the code in a _ems string inside a class that inherits from `ipyreact.ReactWidget`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c70385c-638e-482b-80e1-6f27614da867",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton() {\n",
+    "        return < button > X < /button> \n",
+    "    };\"\"\"\n",
+    "\n",
+    "MyExampleWidget()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "35f22aef-50c2-4772-b9fa-44f5fb486ee6",
+   "metadata": {},
+   "source": [
+    "now it's time to make some **styling using CSS**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38e8afeb-274c-4c62-99a7-6200fc9c179d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "css_rules = \"\"\"\n",
+    "button {\n",
+    "    background-color: yellow;\n",
+    "    border-radius: 10px;\n",
+    "}\n",
+    "\"\"\"\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
```

Comment:
Is injecting CSS like that the preferred method of CSS styling?  Is there any scoping of CSS rules to a cell?  I don't know, genuinely asking.

Reply:
So if I emit CSS at the top of the notebook, will that affect cells at the bottom of the notebook?

Reply:
Yes, page global. Maybe we should explicitly say that, and when you CSS is global (styling of everything, not a single widget, this does make sense).